numpy/linalg/tests/test_linalg.py

```diff
@@ -1929,6 +1929,38 @@ def test_vector_as_first_and_last_argument(self):
         # the result should be a scalar
         assert_equal(multi_dot([A1d, B, C, D1d]).shape, ())
 
+    def test_basic_function_with_three_arguments_with_out_argument(self):
+        # multi_dot with three arguments uses a fast hand coded algorithm to
+        # determine the optimal order. Therefore test it separately.
+        A = np.random.random((6, 2))
+        B = np.random.random((2, 6))
+        C = np.random.random((6, 2))
+
+        out = np.zeros((6, 2))
+        multi_dot([A, B, C], out=out)
```

Comment:
```suggestion
        ret = multi_dot([A, B, C], out=out)
        assert out is ret
```

Reply:
added in 737ba63fdff4469c76a2e2a68a6936dd01d585f9